berry_server/main.go

```diff
@@ -33,11 +35,36 @@ func (s *server) Drive(stream pb.Driver_DriveServer) error {
 }
 
 func main() {
-	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", 31337))
+	// Listen for GRPC connections.
+	lis, err := net.Listen("tcp", ":"+grpcPort)
 	if err != nil {
 		log.Fatalf("failed to listen: %v", err)
 	}
 	s := grpc.NewServer()
 	pb.RegisterDriverServer(s, &server{})
-	s.Serve(lis)
+
+	// Open broadcast connection.
+	c, err := net.ListenPacket("udp", ":0")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer c.Close()
+
+	dst, err := net.ResolveUDPAddr("udp", "255.255.255.255:8032")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	go func() {
+		for {
```

Comment:
Infinite loop ? we have to implement cancellation when client connects to the server. It will be improvement